core/src/saros/net/xmpp/contact/XMPPContact.java

```diff
@@ -226,19 +205,15 @@ boolean setNickname(String newNickname) {
    * @param status
    * @return true if operation changed general status of contact
    */
-  boolean setResourceStatus(JID fullJid, ContactStatus status) {
+  boolean setResourceStatus(JID fullJid, ContactStatus status, boolean sarosSupport) {
     ContactStatus oldStatus = getStatus();
+    boolean oldSarosSupport = hasSarosSupport();
 
-    Optional<Resource> oldResource =
-        resources.stream().filter(r -> compareRawJid(r.fullJid, fullJid)).findFirst();
-    oldResource.ifPresent(resources::remove);
-    resources.add(
-        new Resource(
-            fullJid,
-            status,
-            oldResource.map(r -> r.features).orElse(EnumSet.noneOf(ResourceFeature.class))));
+    resources.stream().filter(r -> compareRawJid(r.fullJid, fullJid)).map(resources::remove);
```

Comment:
Does removing an element from a collection inside a stream operation work or can this also lead to `ConcurrentModificationExceptions`?

Use `removeIf()` instead?

Also, the new logic removes all matching entries while the old logic only removed the first found entry. Does this make a difference or is there always only one match?

Reply:
There is always only one match, it worked but was a quick change. Changed to removeIf